test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```diff
@@ -1462,6 +1462,42 @@
   (testing "Netscape comment constant has expected value"
     (is (= "Puppet Server Internal Certificate" netscape-comment-value))))
 
+(deftest ensure-no-alt-names-test
+  (let [subject-keys (utils/generate-key-pair 512)
+        subject "new-cert"
+        subject-dn (utils/cn subject)]
+    (testing "when allow-subject-alt-names is false"
+      (testing "rejects alt names that don't match the subject"
+        (let [alt-name-ext {:oid utils/subject-alt-name-oid
+                            :value {:dns-name ["bad-name"]}
+                            :critical false}
+              csr (utils/generate-certificate-request subject-keys subject-dn [alt-name-ext])]
+          (is (thrown+-with-msg?
+               [:kind :disallowed-extension]
+               #".*new-cert.*subject alternative names.*bad-name.*"
+               (ensure-subject-alt-names-allowed! csr false))))
+        (let [alt-name-ext {:oid utils/subject-alt-name-oid
+                              :value {:dns-name ["bad-name" subject]}
+                              :critical false}
+                csr (utils/generate-certificate-request subject-keys subject-dn [alt-name-ext])]
+          (is (thrown+-with-msg?
+               [:kind :disallowed-extension]
+               #".*new-cert.*subject alternative names.*bad-name.*"
+               (ensure-subject-alt-names-allowed! csr false)))))
+      (testing "allows a single alt name matching the subject"
```

Comment:
would it be worth testing that the debug message is logged when `allow-subject-alt-names` is false and not when it's true? or nah?